pkg/util/common.go

```diff
@@ -1347,25 +1304,32 @@ func DeleteCustomResourceDefinition(apiExtensionClient *apiextensionsclient.Clie
 }
 
 // WaitUntilPodsAreReady will wait for the pods to be ready
-func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) (bool, error) {
-	// timer starts the timer for timeoutInSeconds. If the task doesn't completed, return error
-	timeout := time.NewTimer(time.Duration(timeoutInSeconds) * time.Second)
-	// ticker starts and execute the task for every n intervals
-	ticker := time.NewTicker(10 * time.Second)
-	for {
-		select {
-		case <-timeout.C:
-			ticker.Stop()
-			return false, fmt.Errorf("[NS: %s | Label: %s] the pods weren't ready - timing out after %d seconds", namespace, labelSelector, timeoutInSeconds)
-		case <-ticker.C:
-			ready, err := IsPodReady(clientset, namespace, labelSelector)
-			if err != nil || ready {
-				timeout.Stop()
-				ticker.Stop()
-				return ready, err
-			}
+func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) error {
```

Comment:
IMO I think using Timer and Ticker, like what was there before, is easier to understand and follow.  Take a timeout from user/default and set a timer (close with defer timer.Close).  We can check however often we want via a ticker, and if timer fires we should do final check of running pods and then return error if pods still aren't running.

Reply:
Previously, there was an issue when timer was set to less than ticker time.  With a ticker and timer, we also weren't checking on the first tick.  I don't understand why both a timer and a ticker are needed, why use two clocks when sum of one clock gives you the total time? 😄  ..But I can change it if you'd prefer.

Reply:
If you can make the logic more readable with a single clock that's fine.  Generally timeout functions work with a total allowed time (timeout) and an interval for checking status.  Timeout is configurable, and sometimes the interval is too.  2 clocks allows the separation of that logic very well even if our original impl had issues, but it can be done in other ways.  The way the logic is now, the tick time isn't regular.  It is timeout/10, which means tick time could be anything.

So, however you want to impl the logic is fine so long as timeout is total time waited and tick time is a consistent interval.